src/config/enterprise.rs

```diff
@@ -0,0 +1,426 @@
+use std::{
+    env,
+    fmt::{Display, Formatter},
+};
+
+use http::Request;
+use hyper::{header::LOCATION, Body, StatusCode};
+use serde::{Deserialize, Serialize};
+use tokio::{
+    select,
+    time::{sleep, Duration},
+};
+use url::{ParseError, Url};
+
+use super::{
+    load_source_from_paths, process_paths, ComponentKey, Config, ConfigPath, OutputId, SinkOuter,
+    SourceOuter,
+};
+use crate::{
+    common::datadog::get_api_base_endpoint,
+    http::{HttpClient, HttpError},
+    signal::{SignalRx, SignalTo},
+    sinks::datadog::metrics::DatadogMetricsConfig,
+    sources::{host_metrics::HostMetricsConfig, internal_metrics::InternalMetricsConfig},
+};
+use vector_core::config::proxy::ProxyConfig;
+
+static HOST_METRICS_KEY: &str = "#datadog_host_metrics";
+static INTERNAL_METRICS_KEY: &str = "#datadog_internal_metrics";
+static DATADOG_METRICS_KEY: &str = "#datadog_metrics";
+
+const DATADOG_REPORTING_PRODUCT: &str = "Datadog Observability Pipelines";
+static DATADOG_REPORTING_PATH_STUB: &str = "/api/unstable/observability_pipelines/configuration";
```

Comment:
Note: This endpoint is likely to change over time.

Basic redirection is handled in this PR, but if we wind up moving to a GraphQL interface per recent discussions, there will be a need for a hard upgrade at some point.